src/main/java/in/projecteka/gateway/clients/ClientError.java

```diff
@@ -25,14 +25,14 @@ public static ClientError unableToConnect() {
                 new ErrorRepresentation(new Error(UNKNOWN_ERROR_OCCURRED, CANNOT_PROCESS_REQUEST_TRY_LATER)));
     }
 
-    public static ClientError hipIdMissing() {
+    public static ClientError idMissingInHeader(String id) {
         return new ClientError(INTERNAL_SERVER_ERROR,
-                new ErrorRepresentation(new Error(UNKNOWN_ERROR_OCCURRED, "X-HIP-ID missing on headers")));
+                new ErrorRepresentation(new Error(UNKNOWN_ERROR_OCCURRED, "Id missing in headers : " + id)));
```

Comment:
Reusing the same method is fine, but if error message is explicit it would be easier for the error receiver to reason about it, instead of having generic `id` is missing.

Reply:
We are also passing the which id is missing. I guess that should help.

Reply:
then you say String.format("%s missing in headers", id) no?